core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -87,6 +88,7 @@ import org.apache.spark.storage.{BlockId, DiskBlockObjectWriter}
  * - Users are expected to call stop() at the end to delete all the intermediate files.
  */
 private[spark] class ExternalSorter[K, V, C](
+    context: TaskContext,
```

Comment:
Also, having it default doesn't buy us much, since we'd still need to update the other call-sites because positional arguments have now changed. I suppose we could also swap the ordering of arguments. I'd like to just leave this as-is, though; it's not a big deal to pass it explicitly at the one or two call sites.